crates/interpreter/src/instructions/stack.rs

```diff
@@ -33,6 +33,38 @@ pub fn push<const N: usize, WIRE: InterpreterTypes, H: Host + ?Sized>(
     _host: &mut H,
 ) {
     gas!(interpreter, gas::VERYLOW);
+
+    // Special optimization for PUSH2 followed by JUMP
+    //
+    // PUSH2 will push the jump destination onto the stack
+    if N == 2 {
+        // Try to read the immediate value (jump destination)
+        // TODO: check what happens if there is less than N bytes to read
+        // TODO: The if statement is here assuming it will read as much as it can
+        // TODO: so it is a boundary check
+        let imm = interpreter.bytecode.read_slice(N);
+        if imm.len() == N {
```

Comment:
This check can be removed, all bytecodes are padded with zeroes to ensure that push always has immediate and that STOP is always last opcode